Add sb-contrib plugin to spotbugs configuration

[fb-contrib](https://github.com/mebigfatguy/fb-contrib) is a plugin for FindBugs and SpotBugs.

There are many detectors in this plugin that are not good enough to be accepted by the maintainers of SpotBugs.
Most of them are questionable and should be excluded, but there are a few good detectors.

For example:
https://github.com/checkstyle/checkstyle/blob/ba568256b4dad3e0d0537b747d2d8b62f9f27ce5/src/main/java/com/puppycrawl/tools/checkstyle/Checker.java#L242-L248
```
[ERROR] Low: Method Checker.getExternalResourceLocations() appears to call the same method on the same object redundantly
[com.puppycrawl.tools.checkstyle.Checker] At Checker.java:[line 245] PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS
```
And there is an issue: `getExternalResourceLocations` is not a simple getter, it does some work under the hood.
Such method call should be extracted from the loop.
 
Another one:
https://github.com/checkstyle/checkstyle/blob/ba568256b4dad3e0d0537b747d2d8b62f9f27ce5/src/main/java/com/puppycrawl/tools/checkstyle/JavaAstVisitor.java#L1909-L1910
```
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.JavaAstVisitor.visitSuperSuffixDot(JavaLanguageParser$SuperSuffixDotContext) uses immediate execution of a block of code that is often not used
[com.puppycrawl.tools.checkstyle.JavaAstVisitor] At JavaAstVisitor.java:[line 1910] OI_OPTIONAL_ISSUES_USES_IMMEDIATE_EXECUTION
```

There should be `.orElseGet`, and this is the #10906

33 more errors reported by JDK11 build:
```
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.ConfigurationLoader creates and initializes a collection but never reads or gains information from it [com.puppycrawl.tools.checkstyle.ConfigurationLoader] In ConfigurationLoader.java WOC_WRITE_ONLY_COLLECTION_FIELD
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator$CliOptions] In JavadocPropertiesGenerator.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.JavadocPropertiesGenerator$CliOptions] In JavadocPropertiesGenerator.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.Main$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.Main$CliOptions] In Main.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.Main$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.Main$CliOptions] In Main.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.Main$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.Main$CliOptions] In Main.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.Main$CliOptions defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.Main$CliOptions] In Main.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.PropertyCacheFile$ExternalResource defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.PropertyCacheFile$ExternalResource] At PropertyCacheFile.java:[line 398] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.PropertyCacheFile$ExternalResource defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.PropertyCacheFile$ExternalResource] At PropertyCacheFile.java:[line 397] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.api.AbstractCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.api.AbstractCheck$FileContext] In AbstractCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.api.AbstractCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.api.AbstractCheck$FileContext] At AbstractCheck.java:[line 303] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck$FileContext] In AbstractFileSetCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck$FileContext] At AbstractFileSetCheck.java:[line 251] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details] At RightCurlyCheck.java:[line 537] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details] At RightCurlyCheck.java:[line 539] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details] At RightCurlyCheck.java:[line 538] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.blocks.RightCurlyCheck$Details] At RightCurlyCheck.java:[line 540] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.DeclarationOrderCheck$ScopeState defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.DeclarationOrderCheck$ScopeState] At DeclarationOrderCheck.java:[line 486] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.DeclarationOrderCheck$ScopeState defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.DeclarationOrderCheck$ScopeState] At DeclarationOrderCheck.java:[line 489] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate] In FinalLocalVariableCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate] In FinalLocalVariableCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$FinalVariableCandidate] At FinalLocalVariableCheck.java:[line 889] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$ScopeData defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$ScopeData] In FinalLocalVariableCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$ScopeData defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck$ScopeData] At FinalLocalVariableCheck.java:[line 846] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport] At CustomImportOrderCheck.java:[line 1399] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport] At CustomImportOrderCheck.java:[line 1400] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck$RuleMatchForImport] At CustomImportOrderCheck.java:[line 1401] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck$FileContext] In AbstractJavadocCheck.java FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck$FileContext defines fields that are used only as locals [com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck$FileContext] At AbstractJavadocCheck.java:[line 412] FCBL_FIELD_COULD_BE_LOCAL
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck creates and initializes a collection but never reads or gains information from it [com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck] In AbstractClassCouplingCheck.java WOC_WRITE_ONLY_COLLECTION_FIELD
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck creates and initializes a collection but never reads or gains information from it [com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck] In AbstractClassCouplingCheck.java WOC_WRITE_ONLY_COLLECTION_FIELD
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck creates and initializes a collection but never reads or gains information from it [com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck] In AbstractClassCouplingCheck.java WOC_WRITE_ONLY_COLLECTION_FIELD
[ERROR] Medium: Class com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck creates and initializes a collection but never reads or gains information from it [com.puppycrawl.tools.checkstyle.checks.metrics.AbstractClassCouplingCheck] In AbstractClassCouplingCheck.java WOC_WRITE_ONLY_COLLECTION_FIELD
```

More errors by drone

```
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.Main.main(String[]) has abnormal exit from finally block [com.puppycrawl.tools.checkstyle.Main] At Main.java:[line 151] AFBR_ABNORMAL_FINALLY_BLOCK_RETURN
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main$CliOptions, List) has abnormal exit from finally block [com.puppycrawl.tools.checkstyle.Main] At Main.java:[line 411] AFBR_ABNORMAL_FINALLY_BLOCK_RETURN
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.execute() has abnormal exit from finally block [com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask] At CheckstyleAntTask.java:[line 308] AFBR_ABNORMAL_FINALLY_BLOCK_RETURN
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.realExecute(String) has abnormal exit from finally block [com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask] At CheckstyleAntTask.java:[line 337] AFBR_ABNORMAL_FINALLY_BLOCK_RETURN
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck.visitToken(DetailAST) is excessively complex, with a cyclomatic complexity of 53 [com.puppycrawl.tools.checkstyle.checks.coding.FinalLocalVariableCheck] In FinalLocalVariableCheck.java CC_CYCLOMATIC_COMPLEXITY
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.getClassFrameWhereViolationIsFound(DetailAST) is excessively complex, with a cyclomatic complexity of 56 [com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck] In RequireThisCheck.java CC_CYCLOMATIC_COMPLEXITY
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.gui.ListToTreeSelectionModelWrapper.updateSelectedPathsFromSelectedRows() has abnormal exit from finally block [com.puppycrawl.tools.checkstyle.gui.ListToTreeSelectionModelWrapper] At ListToTreeSelectionModelWrapper.java:[line 111] AFBR_ABNORMAL_FINALLY_BLOCK_RETURN
[ERROR] Low: Method com.puppycrawl.tools.checkstyle.utils.ScopeUtil.getScope(DetailAST) uses Optional.orElseGet(null) [com.puppycrawl.tools.checkstyle.utils.ScopeUtil] At ScopeUtil.java:[line 71] OI_OPTIONAL_ISSUES_USES_ORELSEGET_WITH_NULL
[ERROR] Low: Method com.puppycrawl.tools.checkstyle.utils.ScopeUtil.getScopeFromMods(DetailAST) uses Optional.orElseGet(null) [com.puppycrawl.tools.checkstyle.utils.ScopeUtil] At ScopeUtil.java:[line 84] OI_OPTIONAL_ISSUES_USES_ORELSEGET_WITH_NULL
```

--- a/pom.xml
+++ b/pom.xml
@@ -495,8 +495,8 @@
             <spotbugsXmlOutputDirectory>target/spotbugsreports</spotbugsXmlOutputDirectory>
             <plugins>
               <plugin>
-                <groupId>com.mebigfatguy.fb-contrib</groupId>
-                <artifactId>fb-contrib</artifactId>
+                <groupId>com.mebigfatguy.sb-contrib</groupId>
+                <artifactId>sb-contrib</artifactId>
                 <version>7.4.7</version>
               </plugin>
             </plugins>

